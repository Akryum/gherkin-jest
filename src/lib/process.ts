import Parser from './parser';

export default function process(src: string, filename: string) {
  const parser = new Parser(src, filename);
  const feature = parser.parse();

  return `
const {cucumber} = require("gherkin-jest");

describe("Feature: " + ${JSON.stringify(feature.name)}, () => {${feature.scenarios.map((scenario) => `
  it("supports the scenario: " + ${JSON.stringify(scenario.name)}, () => {
    ${mapRules('given', scenario.given)}
    ${mapRules('when', scenario.when)}
    ${mapRules('then', scenario.then)}
  });`).join('')}
});`;
}

function mapRules(fn: string, rules: string[]) {
  return rules.map((rule) => `cucumber.${fn}(${JSON.stringify(rule)});`).join('\n');
}
