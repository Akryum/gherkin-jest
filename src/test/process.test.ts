import process from '../lib/process';
import { readFileSync } from 'fs';

describe('process', () => {
  it('should return the correct JS', () => {
    const js = process(readFileSync(__dirname + '/test.feature', 'utf8'), 'test.js');

    expect(js).toMatchSnapshot();
  })
})