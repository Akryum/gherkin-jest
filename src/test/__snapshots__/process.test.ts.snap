// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process should return the correct JS 1`] = `
"const {cucumber} = require(\\"stucumber\\"); const _cucumber = cucumber.clone();describe(\\"Feature: using feature files in jest\\", () => {const feature = {\\"name\\":\\"using feature files in jest\\",\\"annotations\\":[{\\"name\\":\\"someAttribute\\",\\"arguments\\":[]}],\\"meta\\":{}, \\"filename\\": __filename};beforeAll(() => {_cucumber.defineRule(\\"I {op:word} {a:int} and {b:int}\\", (world, params) => Promise.resolve().then(() => _cucumber.rule(world, \\"I have numbers <a> and <b>\\", null, params)).then(() => _cucumber.rule(world, \\"I <op> them\\", null, params)));_cucumber.enterFeature(feature);
      });afterAll(() => _cucumber.exitFeature(feature));it(\\"a simple arithmetic test\\", () => {const world = _cucumber.createWorld();const scenario = {\\"name\\":\\"a simple arithmetic test\\",\\"annotations\\":[{\\"name\\":\\"some-other-attribute\\",\\"arguments\\":[]}],\\"meta\\":{}, \\"feature\\": feature, \\"steps\\": [{\\"name\\":\\"I have numbers 3 and 4\\",\\"line\\":13},{\\"name\\":\\"I add them\\",\\"line\\":14},{\\"name\\":\\"I get 7\\",\\"line\\":15}]};return _cucumber.enterScenario(world, scenario).then(() => _cucumber.rule(world, \\"I have numbers 3 and 4\\", null)).then(() => _cucumber.rule(world, \\"I add them\\", null)).then(() => _cucumber.rule(world, \\"I get 7\\", null)).then(() => _cucumber.exitScenario(world, scenario));});it(\\"a more complex test\\", () => {const world = _cucumber.createWorld();const scenario = {\\"name\\":\\"a more complex test\\",\\"annotations\\":[],\\"meta\\":{}, \\"feature\\": feature, \\"steps\\": [{\\"name\\":\\"I have numbers 6 and 4\\",\\"line\\":18},{\\"name\\":\\"I subtract them\\",\\"line\\":19},{\\"name\\":\\"I get 2\\",\\"line\\":20}]};return _cucumber.enterScenario(world, scenario).then(() => _cucumber.rule(world, \\"I have numbers 6 and 4\\", null)).then(() => _cucumber.rule(world, \\"I subtract them\\", null)).then(() => _cucumber.rule(world, \\"I get 2\\", null)).then(() => _cucumber.exitScenario(world, scenario));});it(\\"a more complex test\\", () => {const world = _cucumber.createWorld();const scenario = {\\"name\\":\\"a more complex test\\",\\"annotations\\":[],\\"meta\\":{}, \\"feature\\": feature, \\"steps\\": [{\\"name\\":\\"I have numbers 8 and 2\\",\\"line\\":18},{\\"name\\":\\"I divide them\\",\\"line\\":19},{\\"name\\":\\"I get 4\\",\\"line\\":20}]};return _cucumber.enterScenario(world, scenario).then(() => _cucumber.rule(world, \\"I have numbers 8 and 2\\", null)).then(() => _cucumber.rule(world, \\"I divide them\\", null)).then(() => _cucumber.rule(world, \\"I get 4\\", null)).then(() => _cucumber.exitScenario(world, scenario));});it(\\"using declared rules\\", () => {const world = _cucumber.createWorld();const scenario = {\\"name\\":\\"using declared rules\\",\\"annotations\\":[],\\"meta\\":{}, \\"feature\\": feature, \\"steps\\": [{\\"name\\":\\"I add 3 and 4\\",\\"line\\":28},{\\"name\\":\\"I get 7\\",\\"line\\":29}]};return _cucumber.enterScenario(world, scenario).then(() => _cucumber.rule(world, \\"I add 3 and 4\\", null)).then(() => _cucumber.rule(world, \\"I get 7\\", null)).then(() => _cucumber.exitScenario(world, scenario));});});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRlc3QuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUMsNEVBQ1EsME5BS0Qsd0dBQ0YsaUVBREUsYUFFRixrREFGRSxHQUxDO0FBQUEsd0RBVUcsOFlBQ0YscURBREUsYUFFSCx5Q0FGRyxhQUdILHNDQUhHLHlEQUtRLDJWQUNWLHFEQURVLGFBRVgsOENBRlcsYUFHWCxzQ0FIVyxrWkFDVixxREFEVSxhQUVYLDRDQUZXLGFBR1gsc0NBSFcseURBVVIsK1NBQ0YsNENBREUsYUFFSCxzQ0FGRyx5REF6QkgiLCJmaWxlIjoidGVzdC5qcyJ9"
`;
